Program.cs

```diff
@@ -1,410 +1,427 @@
-                }
+                case (int)enumMenuMain.RegistrySales:
+                    RegistryInvoice();
+                    break;
+                case (int)enumMenuMain.Reports:
+                    MenuReport();
+
+                    if (!int.TryParse(Console.ReadLine(), out option))
+                    {
+                        return;
+                    }
+
+                    switch (option)
+                    {
+                        case (int)enumMenuReport.Sales:
+                            ReportSales();
+                            break;
+                        case (int)enumMenuReport.CustomersAge:
+                            ReportCustomersAge();
+                            break;
+                        case (int)enumMenuReport.CustomersBirthdayMonth:
+                            ReportCustomersBirthay();
+                            break;
+                        case (int)enumMenuReport.ProductsStock:
+                            ReportProductStock();
+                            break;
+                        default:
+                            Console.WriteLine("Ingrese una opcion correcta");
+                            MenuMain();
+                            ProcessMain();
+                            break;
+                    }
+                    break;
+                default:
+                    Console.WriteLine("Ingrese una opcion correcta");
+                    MenuMain();
+                    ProcessMain();
+                    break;
             }
+        }
+
+        private static void MenuMain()
+        {
+            Console.WriteLine("1. Mantenimiento");
+            Console.WriteLine("2. Registro de Venta");
+            Console.WriteLine("3. Reportes");
+        }
 
-            void MenuPrincipal()
+        private static void MenuMaintenance()
+        {
+            Console.WriteLine("1. Cliente");
+            Console.WriteLine("2. Producto");
+            Console.WriteLine("3. Vendedor");
+            Console.WriteLine("9. Menu Principal");
+        }
+
+        private static void MenuReport()
+        {
+            Console.WriteLine("1. Ventas");
+            Console.WriteLine("2. Clientes mayores 55");
+            Console.WriteLine("3. Clientes cumplan este mes");
+            Console.WriteLine("4. Productos por debajo stock");
+        }
+
+        private static void RegistryCustomer()
+        {
+            Console.WriteLine("Codigo Cliente: ");
+            if (!int.TryParse(Console.ReadLine(), out int idCustomer))
             {
-                Console.WriteLine("1. Mantenimiento");
-                Console.WriteLine("2. Registro de Venta");
-                Console.WriteLine("3. Reportes");
+                return;
             }
 
-            void MenuMaintenance()
+            Console.WriteLine("Nombre: ");
+            string nameCustomer = Console.ReadLine();
+
+            Console.WriteLine("Fecha de Nacimiento (MM/dd/yyyy): ");

```

Comment:
As I mentioned, you could make a method to do this.
bool GetRequesTValue(string message, out Enum result)
{
  Console.WriteLine(message);
  string response = Console.ReadLine();
  return Enum.TryParse( response, false, out result);//here I don't remember exactly the syntax
}
and the just use it

GetRequesTValue("enter a value", out MyEnum myEnum);
something like this.